### ISSUE TITLE
1.14.4-6.0.2 - Incompatibility with the mod "Silent's Mechanisms"

### ISSUE THREAD

**Issue by elowanut** [jaquadro/StorageDrawers#734]
## Versions

**Storage Drawers**: 1.14.4-6.0.2
**Silent's Mechanisms**: 1.14.4-0.5.2+28 (also tested 25)
**Forge**: 1.14.4-28.0.102 (also does it on 91)

## Expected Behavior

When using the "compacting drawer" from this mod, right click on the drawer to put in, say, a stack of iron ingots. The expected result is that the drawer now contains 64 iron ingots, and displays the icon for ingots/nuggets/blocks on its front face.

## Actual Behavior
The game instantly CTDs, eventually loading up the launcher with a crash report. Note that the crash occurs with any item, even ones that aren't typically used in compacting drawers (e.g. items with no alternate forms, like silk-touched ores, or logs). Note that this only occurs with Silent's Mechanisms loaded. Other mods, including Silent's Lib / Gems / Gear are fine.

## Links/Images

Crash report here:
https://www.dropbox.com/s/znxn0wajgzdfr9v/crash-2019-09-10_11.58.27-client.txt?dl=0

## Steps to Reproduce the Problem

1. Load up a forge modded version of minecraft with silent lib / mechanisms, and the Storage Drawers mod.
2. Start a fresh world, place down a compacting drawer.
3. Right click (try to put items into) the drawer with any item.

## Notes

I also filed a bug report on the Silent's Mechanisms GitHub. Silent suggested that it should be addressed here.

**Comment by SilentChaos512** [comment]
One possible solution is to change `CompactingHelper.findAllMatchingRecipes` to get only crafting recipes. Something like this:

```java
for (ICraftingRecipe recipe : world.getRecipeManager().getRecipes(IRecipeType.CRAFTING, crafting, world)) {
    if (recipe.matches(crafting, world)) {
        ItemStack result = recipe.getCraftingResult(crafting);
        if (!result.isEmpty())
            candidates.add(result);
    }
}
```

Issue on my repo, just for reference: https://github.com/SilentChaos512/Silents-Mechanisms/issues/15

**Comment by jaquadro** [comment]
Thanks for pointing out (more) correct usage of the new RecipeManager.  Should be fixed in 6.0.3.
